pulsar-flink-connector/src/main/java/org/apache/flink/streaming/connectors/pulsar/internal/PulsarMetadataReader.java

```diff
@@ -241,17 +245,6 @@ public void deleteTopic(String topicName) throws PulsarAdminException {
         try {
             PartitionedTopicInternalStats partitionedInternalStats =
                     admin.topics().getPartitionedInternalStats(topicName);
-            final Optional<PersistentTopicInternalStats> any =
```

Comment:
My observation here is that, we want to call the delete even when there are cursors present in the topic. Two places called this method in PulsarCatalogSupport:

``` createTable()
        // use pulsar schema to store generic table information
        try {
            SchemaInfo schemaInfo = TableSchemaHelper.generateSchemaInfo(table.toProperties());
            pulsarMetadataReader.uploadSchema(topicName, schemaInfo);
        } catch (Exception e) {
            // delete topic if table info cannot be persisted
            try {
                pulsarMetadataReader.deleteTopic(topicName);
            } catch (PulsarAdminException ex) {
                // do nothing
            }
            e.printStackTrace();
            throw new CatalogException("Can't store table metadata");
        }
```

and dropTable(). In both two methods we want the topics deleted whether we have cursors or not.

Reply:
change to `delete(String topicName, boolean force)`

Reply:
The pulsarMetadataReader uses deletePartitionedTopic(topicName, true) and will call delete(topicName, true) if topic is not found. Is any reason we need to use delete(String topicName, boolean force) instead of deletePartitionedTopic(String topicName, boolean force) here ?